### PR TITLE
JTE/PKFE-42

### PR DESCRIPTION
**Pull request by justinnas**
- Moved editor toolbar groups into seperate files;
- Moved BaseLayout into routes;
- Moved Sidebar to a seperate file;
- Made Sidebar icons clickable;
- Made it so the icons would be highlighted based on page being visited;
- Fixed other minor errors.

### CODE REVIEW

File: app/front-end/src/components/buttons/IconTitleButton.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ interface Props {
   height?: string;
   borderRadius?: string;
   onClick?: () => void;
+  isActive?: boolean;
```
Comment (mantvydasdeltuva) [inline app/front-end/src/components/buttons/IconTitleButton.tsx:11]: Also add disabled for unusable buttons like macros to not be available

File: app/front-end/src/components/layouts/baseLayout.tsx [hunk 2]
```diff
@@ -1,12 +1,6 @@
-import { IconTitleButton } from '@/components/buttons/IconTitleButton';
 import { SettingsDialog, ShortcutsDialog } from '@/components/dialogs';
+import { Sidebar } from '@/components/sidebar/sidebar';
```
Comment (mantvydasdeltuva) [inline app/front-end/src/components/layouts/baseLayout.tsx:2]: Add `index.ts` to remove redundancy in pathing

**Review by mantvydasdeltuva — CHANGES_REQUESTED** [review]

File: app/front-end/src/components/layouts/baseLayout.tsx (hunk 2 above)
Comment (justinnas) [reply app/front-end/src/components/layouts/baseLayout.tsx]: Done!

File: app/front-end/src/components/buttons/IconTitleButton.tsx (hunk 1 above)
Comment (justinnas) [reply app/front-end/src/components/buttons/IconTitleButton.tsx:10]: Done!

**Comment by mantvydasdeltuva** [comment]
Approved!

**Review by mantvydasdeltuva — APPROVED** [review]
Approved!

**Pull request merged**